src/managers/BaseManager.ts

```diff
@@ -64,21 +64,18 @@ export abstract class BaseManager {
     try {
       isKeySuccessfullyLocked = await this.storage.lockKey(context.key);
     } catch (keyLockError: unknown) {
-      this.logger.error(
+      this.logger.info(
         `Error occurred while trying to lock key "${context.key}". Reason: ${
           (keyLockError as Error).message
         }. Running executor`
       );
-
-      return runExecutor(context.executor);
```

Comment:
Is there any test which checks that no executor is running?

Reply:
This particular line is a bug. Code prematurely exits without setting any value for the key.